src/main/java/net/fabricmc/loader/transformer/ClassStripper.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.transformer;
+
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Strips the specified interfaces, fields and methods from a class.
+ */
+public class ClassStripper extends ClassVisitor {
```

Comment:
ClassEnvironmentRemover?

Reply:
Actually, no. ClassEnvironmentStripper

Reply:
This class is not strictly related to environment stripping which is why I am reluctant to put it there. It could be used to strip stuff from classes based on other conditions.